Closing: my "3.22 is enough" analysis missed a silent CMake-version requirement.

`source/MRCuda/CMakeLists.txt` configures `CUDA_STANDARD 20`, and the image's NVCC is 12.6.85 (`-- The CUDA compiler identification is NVIDIA 12.6.85`). CMake 3.22's NVCC-flag table has no entry for that combination:

```
CMake Error in /__w/MeshLib/MeshLib/build/Release/CMakeFiles/CMakeTmp/CMakeLists.txt:
  Target "cmTC_c5d22" requires the language dialect "CUDA20" (with compiler extensions),
  but CMake does not know the compile flags to use to enable it.
Call Stack:
  source/MRCuda/CMakeLists.txt:4 (project)
```

(Seen on run [24821677923](https://github.com/MeshInspector/MeshLib/actions/runs/24821677923/job/72650815503).)

So the upgrade to a newer CMake was functionally required after all — just not captured in any `cmake_minimum_required(...)` line in the tree, which is what I checked. The right framing for the apt.kitware.com reliability problem is PR #5962 (swap to GitHub Releases tarball for Kitware's same CMake). I'll push that through instead.

--- a/cmake/Modules/ConfigureCuda.cmake
+++ b/cmake/Modules/ConfigureCuda.cmake
@@ -11,12 +11,7 @@ IF(NOT CUDAToolkit_FOUND AND NOT CUDA_FOUND)
       # For our VS2022 CI:
       set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -allow-unsupported-compiler -D_ALLOW_COMPILER_AND_STL_VERSION_MISMATCH")
     ELSE()
-      # CMake's NVCC flag-table entry for C++20 was added in the CMake 3.25.2
-      # patch release (gated on NVCC >= 12.0); earlier 3.25.x / 3.24 / 3.22
-      # etc. fail at CUDA project detection with "does not know the compile
-      # flags to use to enable it." Ubuntu 22.04 jammy ships 3.22.x through
-      # apt, well below the threshold, so fall back to C++17 there. MeshLib's
-      # CUDA code compiles cleanly under either standard.
+      # https://cmake.org/cmake/help/latest/release/3.25.html#id2
       IF(CMAKE_VERSION VERSION_LESS 3.25.2)
         set(CMAKE_CUDA_STANDARD 17)
       ELSE()

--- a/cmake/Modules/ConfigureCuda.cmake
+++ b/cmake/Modules/ConfigureCuda.cmake
@@ -11,13 +11,13 @@ IF(NOT CUDAToolkit_FOUND AND NOT CUDA_FOUND)
       # For our VS2022 CI:
       set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -allow-unsupported-compiler -D_ALLOW_COMPILER_AND_STL_VERSION_MISMATCH")
     ELSE()
-      # CMake's NVCC flag-table entry for C++20 was added in CMake 3.26 (gated
-      # on NVCC >= 12.0). On older CMake -- notably Ubuntu 22.04 jammy's
-      # apt-supplied 3.22 -- using CUDA_STANDARD=20 fails at CUDA project
-      # detection with "does not know the compile flags to use to enable it".
-      # Fall back to C++17 in that case; MeshLib's CUDA code compiles under
-      # either standard.
-      IF(CMAKE_VERSION VERSION_LESS 3.26)
+      # CMake's NVCC flag-table entry for C++20 was added in the CMake 3.25.2
+      # patch release (gated on NVCC >= 12.0); earlier 3.25.x / 3.24 / 3.22
+      # etc. fail at CUDA project detection with "does not know the compile
+      # flags to use to enable it." Ubuntu 22.04 jammy ships 3.22.x through
+      # apt, well below the threshold, so fall back to C++17 there. MeshLib's
+      # CUDA code compiles cleanly under either standard.
+      IF(CMAKE_VERSION VERSION_LESS 3.25.2)
         set(CMAKE_CUDA_STANDARD 17)
       ELSE()
         set(CMAKE_CUDA_STANDARD 20)

--- a/cmake/Modules/ConfigureCuda.cmake
+++ b/cmake/Modules/ConfigureCuda.cmake
@@ -11,7 +11,17 @@ IF(NOT CUDAToolkit_FOUND AND NOT CUDA_FOUND)
       # For our VS2022 CI:
       set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -allow-unsupported-compiler -D_ALLOW_COMPILER_AND_STL_VERSION_MISMATCH")
     ELSE()
-      set(CMAKE_CUDA_STANDARD 20)
+      # CMake's NVCC flag-table entry for C++20 was added in CMake 3.26 (gated
+      # on NVCC >= 12.0). On older CMake -- notably Ubuntu 22.04 jammy's
+      # apt-supplied 3.22 -- using CUDA_STANDARD=20 fails at CUDA project
+      # detection with "does not know the compile flags to use to enable it".
+      # Fall back to C++17 in that case; MeshLib's CUDA code compiles under
+      # either standard.
+      IF(CMAKE_VERSION VERSION_LESS 3.26)
+        set(CMAKE_CUDA_STANDARD 17)
+      ELSE()
+        set(CMAKE_CUDA_STANDARD 20)
+      ENDIF()
       find_package(CUDAToolkit 12 REQUIRED)
     ENDIF()
     set(CMAKE_CUDA_STANDARD_REQUIRED ON)

